### DISCUSSION TITLE
Couldn't enable/disable the hard limit settings. Couldn't find a setting with key $22 to update.

### DISCUSSION THREAD

**Discussion by sound-artefacts** [winder/Universal-G-Code-Sender#2223] [Help]
Hi, I am new in the field and I don't know how to solve this problem. I am using an Arduino Uno as a controller and I think I have uploaded the grbl firmware correctly. Can someone direct me on what I am doing wrong regarding the Error mentioned in the title? The Error comes out in the setup wizard when I am trying to enable homing in UGS.

**Comment by breiler** [comment]
Which version of UGS are you running?
What version of GRBL is your controller using?

Can you copy'n'paste the whole communication from the console after you have connected.

**Reply by sound-artefacts** [reply]
Hi, the communication from the console writes as follows:

```
 **** Connected to /dev/ttyACM0 @ 115200 baud ****
i
Grbl 0.9j ['$' for help]
>>> $$
$1 = 25    (step idle delay, msec)
$0 = 10    (step pulse, usec)
$2 = 0    (step port invert mask:00000000)
$3 = 0    (dir port invert mask:00000000)
$4 = 0    (step enable invert, bool)
$5 = 1    (limit pins invert, bool)
$6 = 0    (probe pin invert, bool)
$10 = 3    (status report mask:00000011)
$11 = 0.010    (junction deviation, mm)
$12 = 0.002    (arc tolerance, mm)
$13 = 0    (report inches, bool)
$20 = 0    (soft limits, bool)
$21 = 1    (hard limits, bool)
$22 = 1    (homing cycle, bool)
$23 = 3    (homing dir invert mask:00000011)
$24 = 25.000    (homing feed, mm/min)
$26 = 250    (homing debounce, msec)
$25 = 500.000    (homing seek, mm/min)
$27 = 1.000    (homing pull-off, mm)
$101 = 40.000    (y, step/mm)
$100 = 40.000    (x, step/mm)
$102 = 200.000    (z, step/mm)
$110 = 500.000    (x max rate, mm/min)
$111 = 500.000    (y max rate, mm/min)
$112 = 500.000    (z max rate, mm/min)
$120 = 10.000    (x accel, mm/sec^2)
$121 = 10.000    (y accel, mm/sec^2)
$122 = 10.000    (z accel, mm/sec^2)
$130 = 1000.000    (x max travel, mm)
$131 = 1000.000    (y max travel, mm)
$132 = 1000.000    (z max travel, mm)
>>> $G
ok
ok
[G0 G54 G17 G21 G90 G94 M0 M5 M9 T0 F0. S0.]
>>> $X
[Caution: Unlocked]
ok
```


I am guessing $22 homing cycle =1 means it is enabled. The error comes out during setup when I am brought to the limit switches window and enable homing window and the error persists after finishing the setup, as soon as I try to home the machine. The machine is a styrofoam cutter and the z-axis is a turntable.

**Comment by sound-artefacts** [comment]
Hi Mr.Breiler, I am not sure, I downloaded grbl from here[ ](https://github.com/grbl/grbl) so it should be this latest version, as for the UGS it is version 2.0.14 for Linux. I currently do not have access to the system but I will post the communication as soon as possible.

Thank you
